controllers/buildpipeline/buildpipeline_adapter.go

```diff
@@ -65,8 +65,9 @@ func NewAdapter(pipelineRun *tektonv1.PipelineRun, component *applicationapiv1al
 // to the build PipelineRun being processed exists. Otherwise, it will create a new pipeline Snapshot.
 func (a *Adapter) EnsureSnapshotExists() (controller.OperationResult, error) {
 	if !h.HasPipelineRunSucceeded(a.pipelineRun) {
-		if h.HasPipelineRunFinished(a.pipelineRun) {
+		if h.HasPipelineRunFinished(a.pipelineRun) || a.pipelineRun.GetDeletionTimestamp() != nil {
```

Comment:
If one PLR works well but has not finished, is it possible the prune mechanism tries to delete it? If yes, an in-progess PLR might be deleted by this code as well. I mean when does the prune mechanism delete the PLR? Only after the PLR finishes?

Reply:
It's the purpose of this PR to delete running PLR marked for deletion. If PLR is deleted tekton doesn't change status to non-running, it will be stuck forever in running status. Se we have to "unlock" even running pipeline.

Reply:
makes me think about the purpose of having a finalizer on build PLRs in the first place, it was to ensure that the integration service can create snapshots of successful builds. By this change, we'd allow the pruner to delete the PLR (while it's running) whenever it wants.

Now with this change, the finalizer would only be useful when the pruner tries to delete a successful build PLR for which the integration service hasn't created a Snapshot yet. IIUC

Reply:
If the time between the PLR is started and decided to be deleted is long enough, the change can make sure the build PLR can be deleted correctly.